test/test_requirements.py

```diff
@@ -0,0 +1,127 @@
+)
+DOC_TXT = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), "..", "doc", "requirements.txt")
+)
+DOC_CSV = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), "..", "doc", "requirements.csv")
+)
+
+CACHE_PATH = os.path.join(os.path.expanduser("~"), ".cache", "cve-bin-tool")
+
+CACHE_CSV = os.path.join(CACHE_PATH, "requirements.csv")
+
+HTML_DEP_PATH = os.path.abspath(
+    os.path.join(
+        os.path.dirname(__file__),
+        "..",
+        "cve_bin_tool",
+        "output_engine",
+        "html_reports",
+        "js",
+    )
+)
+HTML_DEP_CSV = os.path.join(HTML_DEP_PATH, "dependencies.csv")
+
+# Dependencies that currently have CVEs
+# Remove from the list once they are updated
+ALLOWED_PACKAGES = ["reportlab"]
+
+
+def get_out_of_sync_packages(csvfile, txtfile):
+
+    out_of_sync_packages = []
+    csv_package_names = []
+
+    with open(csvfile, "r") as csvf, open(txtfile, "r") as txt:
+        csvfl = csv.reader(csvf)
+        next(csvfl)
+        for (vendor, product) in csvfl:
+            csv_package_names.append(product)
+        lines = txt.readlines()
+        for line in lines:
+            txt_package = re.split(">|\\[|;|=|\n", line)[0]
+            if txt_package not in csv_package_names:
+                out_of_sync_packages.append(txt_package)
+
+    return out_of_sync_packages
+
+
+# Test to check if the requirements.csv files are in sync with requirements.txt files
+def test_txt_csv_sync():
+
+    out_of_sync_packages = get_out_of_sync_packages(
+        REQ_CSV, REQ_TXT
+    ) + get_out_of_sync_packages(DOC_CSV, DOC_TXT)
+
+    assert (
+        out_of_sync_packages == []
+    ), f"The CSV file is out of sync! Please add {', '.join(out_of_sync_packages)} in the respective requirements.csv file"
+
+
+def get_cache_csv_data(file):
+
+    data = []
+
+    with open(file, "r") as f:
+        r = csv.reader(f)
+        next(r)
+        for (vendor, product) in r:
+            if file is HTML_DEP_CSV:
+                file_name = f"{HTML_DEP_PATH}/{product}"
+                if not file_name.endswith(".js"):
+                    file_name += ".js"
+                with open(file_name) as f:
+                    file_content = f.read()
+                    html_dep_version = re.search(
+                        r"v([0-9]+\.[0-9]+\.[0-9]+)", file_content
+                    ).group(1)
+                    if product not in ALLOWED_PACKAGES:
+                        data.append((vendor, product, html_dep_version))
+            else:
+                if "_not_in_db" not in vendor and product not in ALLOWED_PACKAGES:
+                    data.append((vendor, product, version(product)))
+
+        return data
+
+
+# Test to check for CVEs in cve-bin-tool requirements/dependencies
+def test_requirements():
+
+    cache_csv_data = (
+        get_cache_csv_data(REQ_CSV)
+        + get_cache_csv_data(DOC_CSV)
+        + get_cache_csv_data(HTML_DEP_CSV)
+    )
+
+    if not os.path.exists(CACHE_PATH):
```

Comment:
What is the point in writing `requirements.csv` into cache? I don't think it's needed, just write it to some other file and drop all this unnecessary logic.

Reply:
I think any new file that is not from the repo is stored in cache.
Yup, it'd work without the cache logic but for consistency's sake I think maybe it's better to keep it this way.

Reply:
On second thought I should probably remove this logic. Too much complexity :(

Reply:
With other jobs cache serves its purpose: in pip cache it somewhat speeds up installs because pip uses that directory (though gains from it deteriorate with time because the cache is immutable but dependencies get new versions), in DB caching case it allows to only download CVE DB once per day (technically once per day in every job because they run in parallel but you get the idea).

But in your case, you ALWAYS write to that cache first, without trying to read. You don't reuse anything, you always get your data from regular files in repo, not cache.